src/Z38/SwissPayment/Text.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Z38\SwissPayment;
+
+use DOMDocument;
+use InvalidArgumentException;
+
+class Text
+{
+    const TEXT_NOT_CH = '/[^A-Za-z0-9 .,:\'\/()?+\-!"#%&*;<>÷=@_$£[\]{}\` ́~àáâäçèéêëìíîïñòóôöùúûüýßÀÁÂÄÇÈÉÊËÌÍÎÏÒÓÔÖÙÚÛÜÑ]+/u';
+    const TEXT_NOT_SWIFT = '/[^A-Za-z0-9 .,:\'\/()?+\-]+/';
+
+    /**
+     * Sanitizes and trims a string to conform to the Swiss character
+     * set.
+     *
+     * @param string|null $input
+     * @param int         $maxLength
+     *
+     * @return string The sanitized string
+     */
+    public static function sanitize($input, $maxLength)
+    {
+        $input = preg_replace('/\s+/', ' ', (string) $input);
+        $input = trim(preg_replace(self::TEXT_NOT_CH, '', $input));
+
+        return function_exists('mb_substr') ? mb_substr($input, 0, $maxLength, 'UTF-8') : substr($input, 0, $maxLength);
+    }
+
+    /**
+     * Sanitizes and trims a string to conform to the Swiss character
+     * set.
+     *
+     * @param string|null $input
+     * @param int         $maxLength
+     *
+     * @return string|null The sanitized string or null if it is empty.
+     */
+    public static function sanitizeOptional($input, $maxLength)
+    {
+        $sanitized = self::sanitize($input, $maxLength);
+
+        return $sanitized !== '' ? $sanitized : null;
+    }
+
+    /**
+     * @internal
+     */
+    public static function sanitizeCountryCode($input)
+    {
+        return preg_replace('/[^A-Z]+/', '', strtoupper($input));
+    }
+
+    /**
+     * @internal
+     */
+    public static function assertOptional($input, $maxLength)
+    {
+        if ($input === null) {
+            return null;
+        }
+
+        return self::assert($input, $maxLength);
+    }
+
+    /**
+     * @internal
+     */
+    public static function assert($input, $maxLength)
+    {
+        return self::assertNotPattern($input, $maxLength, self::TEXT_NOT_CH);
+    }
+
+    /**
+     * @internal
+     */
+    public static function assertIdentifier($input)
+    {
+        $input = self::assertNotPattern($input, 35, self::TEXT_NOT_SWIFT);
+        if ($input[0] === '/' || strpos($input, '//') !== false) {
+            throw new InvalidArgumentException('The identifier contains unallowed slashes.');
+        }
+
+        return $input;
+    }
+
+    /**
+     * @internal
+     */
+    public static function assertCountryCode($input)
```

Comment:
In this assertion, it would be great to check that the country code is valid isn't it?
The full ISO3166 country code list can be retrieved here : https://gist.github.com/vxnick/380904

Reply:
As the developers need to present a list of valid countries the end users anyway, I will leave that open for now.